### PR TITLE
Add test for BalancedMiniBatches

### PR DESCRIPTION
**Pull request by huitseeker**
Side product of some of my other work, but should help.

### CODE REVIEW

File: nd4j-backends/nd4j-tests/src/test/java/org/nd4j/linalg/dataset/BalanceMinibatchesTest.java [hunk 1]
```diff
@@ -30,6 +33,45 @@ public void testBalance() {
...
+
+        int miniBatchSize = 10;
+        DataSetIterator iterator = new IrisDataSetIterator(miniBatchSize, 150);
+        BalanceMinibatches balanceMinibatches = BalanceMinibatches.builder().dataSetIterator(iterator).miniBatchSize(miniBatchSize)
+                .numLabels(iterator.totalOutcomes()).rootDir(new File("minibatches")).rootSaveDir(new File("minibatchessave")).build();
+        balanceMinibatches.balance();
+        DataSetIterator balanced = new ExistingMiniBatchDataSetIterator(balanceMinibatches.getRootSaveDir());
+
+        assert(iterator.resetSupported()); // this is testing the Iris dataset more than anything
+        iterator.reset();
```
Comment (agibsonccc) [inline nd4j-backends/nd4j-tests/src/test/java/org/nd4j/linalg/dataset/BalanceMinibatchesTest.java]: I don't like normal asserts> Please use the junit asserts.

File: nd4j-backends/nd4j-tests/src/test/java/org/nd4j/linalg/dataset/BalanceMinibatchesTest.java [hunk 2]
```diff
@@ -30,6 +33,45 @@ public void testBalance() {
...
+        ArrayList<Double> totalCounts = new ArrayList<Double>(iterator.totalOutcomes());
+        while (iterator.hasNext()) {
```
Comment (agibsonccc) [inline nd4j-backends/nd4j-tests/src/test/java/org/nd4j/linalg/dataset/BalanceMinibatchesTest.java]: In java always use the interface not the implementation. Change ArrayList -> List

File: nd4j-backends/nd4j-tests/src/test/java/org/nd4j/linalg/dataset/BalanceMinibatchesTest.java [hunk 3]
```diff
@@ -30,6 +33,45 @@ public void testBalance() {
 
     }
 
+    public void testMiniBatchBalanced() {
```
Comment (agibsonccc) [inline nd4j-backends/nd4j-tests/src/test/java/org/nd4j/linalg/dataset/BalanceMinibatchesTest.java]: Missing @Test annotation?

**Comment by huitseeker** [comment]
@agibsonccc Fixed, thanks !

**Pull request merged**